lib/internal/url.js

```diff
@@ -874,32 +871,32 @@ class URLSearchParams {
     this[context] = null;
   }
 
-  [util.inspect.custom](recurseTimes, ctx) {
+  [util.inspect.custom](depth, opts) {
     if (!this || !this[searchParams] || this[searchParams][searchParams]) {
       throw new TypeError('Value of `this` is not a URLSearchParams');
     }
 
-    if (typeof recurseTimes === 'number' && recurseTimes < 0)
-      return ctx.stylize('[Object]', 'special');
+    if (typeof depth === 'number' && depth < 0)
+      return opts.stylize('[Object]', 'special');
 
-    const separator = ', ';
-    const innerOpts = Object.assign({}, ctx);
-    if (recurseTimes !== null) {
-      innerOpts.depth = recurseTimes - 1;
+    var separator = ', ';
```

Comment:
I'm not sure it needs a comment. Many optimizations are made without comments describing why they exist, including permanent deopt fixes. Also, the hope is that V8 will (or already has) fix the deopt (perhaps with ignition + turbofan... I don't recall which pipeline the function was being optimized under).

Ideally such permanent deopts would be checked when pushing such code. Perhaps having a Makefile target that runs the parallel tests with `--trace-opt --trace-deopt` and greps for `/aborted|disabled/` would be helpful in achieving this in an automated way.